cpu_scheduling_algorithms/round_robin_scheduling.cpp

```diff
@@ -0,0 +1,242 @@
+     * process and completion time.
+     * 
+     * \param process A process that have been executed
+     * \param completion_time The process execution finish time
+     */
+    ProcessResult(const Process& process, uint32_t completion_time)
+        : Process(process), completion_time(completion_time) {
+        turn_around_time = completion_time - arrival_time;
+        waiting_time = turn_around_time - burst_time;
+    }
+    
+    /**
+     * @brief Compare each member variable.
+     * 
+     * \param p ProcessResult to be compared with
+     * \return true if the processes IS equal
+     * \return false if the processes IS NOT equal
+     */
+    bool operator==(const ProcessResult& p) const {
+        return id == p.id && arrival_time == p.arrival_time &&
+               burst_time == p.burst_time &&
+               completion_time == p.completion_time &&
+               turn_around_time == p.turn_around_time &&
+               waiting_time == p.waiting_time;
+    }
+};
+
+/**
+ * Represent remaining burst time of a process.
+ */
+using BTLeft = uint32_t;
+
+/**
+ * @brief Execute processes based on Round-robin algorithm.
+ * 
+ * \param processes Processes to be executed
+ * \param time_slice Time slice for processes execution
+ * \return Results of each process execution
+ */
+std::vector<ProcessResult> RRExecute(const std::vector<Process>& processes,
+                                     uint32_t time_slice);
+
+/**
+ * @brief Print a table containing process results data.
+ * 
+ * \return ostream inputted ostream
+ */
+std::ostream& operator<<(std::ostream& ostream,
+                         const std::vector<ProcessResult>& results);
+
+/**
+ * @brief Self-test implementations.
+ * 
+ * \returns void
+ */
+void Test();
+
+
+/**
+ * @brief Comparator function for sorting processes.
+ * 
+ * \param p1 Process to be compared
+ * \param p2 Process to be compared
+ * \return 
+ */
+bool CompareAT(const Process& p1, const Process& p2) {
+    return p1.arrival_time < p2.arrival_time;
+}
+
+/**
+ * @brief Check processes that arrive after the given time_elapsed.
+ *
+ * If a process arrive after the give time_elapsed, then the process
+ * will be pushed into the schedule queue and inserted into the
+ * arrived_process set.
+ * 
+ * \param processes Processes that will be checked for arrival
+ * \param arrived_process A set containing processes that has arrived
+ * \param schedule Queue containing pair of process and its remaining burst time
+ * \param time_elapsed Time that has elapsed after processes execution
+ */
+void CheckArriveProcess(const std::vector<Process>& processes,
+                        std::set<uint32_t>& arrived_process,
+                        std::queue<std::pair<Process, BTLeft>>& schedule,
+                        uint32_t time_elapsed);
+
+/**
+ * @brief Entry point of the program.
+ * 
+ * \return 0 on exit
+ */
+int main() {
+    Test();
+    return 0;
+}
+
+void Test() {
```

Comment:
This should probably be documented as the function prototype doesn't contain anything inside and Doxygen may not generate it properly.
```suggestion
/**
 * @brief Self-test implementations
 * @returns void
 */
static void Test() {
```